RPN

Armadz opened the pull request:
"Make a pull request on your branch, too. It doesn't have to be right before you want to merge back in" - Prof Oak 10/21/17

Armadz committed:
Modified Test

* Modified test for powers to reflect changes to power function

Armadz committed:
Changes to Operations

*reworked power function to take advantage of built in python operations

Armadz committed:
Merge branch 'master' into parsing

ProfOak commented on Calculator/operations.py:
```diff
@@ -41,13 +41,10 @@ def ceiling(a):
-    return a
-    # only works for int powers
-    # meed to fix for power of 0
+    if a == 0 and b == 0:
```
Why is it important for these numbers not to be 0?

Armadz replied:
In python 0 ** 0 == 1, I  prefer undetermind

ProfOak replied:
That's fair but in future projects be sure to think about the user that you're intending to use your tool. If you are making this solely for yourself then there is no problem. In our case it's a bit contrived because we're making one of the most ubiquitous programs in the world so people aren't tied to us. 

In the future, make sure you can remember all the cases that are potentially unintuitive to people using your program. You will have to document your intended behavior so your users don't hate you. :-p

Armadz replied:
I honestly thought that's how calculators normally work : selffacepalm:

ProfOak replied:
When you are ready to merge the PR let me know so I can give a proper review/approval. 👍

Armadz committed:
Reverse polish notation evalutor

*work in progress

Armadz committed:
Update to operations created rpn

*power operation is updated to match calculator conventions
*created reverse polish notation evaluation function

Armadz committed:
Updated operation test added rpn test

*operation test brought in line with changes to operations power
*added reverse polish notation test

Armadz committed:
Added power test

*added test to make sure power function works as intended in reverse
polish notation

ProfOak commented on tests/test_rpn.py:
```diff
@@ -5,3 +5,6 @@ class TestRPN(TestCase):
     def test_rpn_eval(self):
         assert rpn.rpn_eval(["3" , "2", "*", "11" , "-"]) == -5.0
+
+    def test_rpn_powers(self):
```
Treat these as unit tests [1]. A unit test is a test for the smallest possible unit of code. Typically your units should be functions/methods, unless your one function does a lot of things. You should try to avoid overloading methods with functionality, if possible, as a good practice. You already have unit tests to make sure that the operators function properly, so you don't need to have extra tests. Remember that a test can have multiple assert statements. So feel free to have a bunch in the `test_rpn_eval` test function!

[1] https://en.wikipedia.org/wiki/Unit_testing

ProfOak committed:
feat(main): add skeleton interface for main function

ProfOak committed:
fix(repl): fix user input

the first user input should exit if the user presses q at the beginning

ProfOak committed:
Merge branch 'parsing' of github.com:ImWalkinHere/Calculator into feat/outline-main-func

Armadz committed:
Merge pull request #5 from ImWalkinHere/feat/outline-main-func

feat(main): add skeleton interface for main function
